api/src/main/kotlin/com/google/devtools/ksp/symbol/KSPropertyDeclaration.kt

```diff
@@ -58,8 +58,30 @@ interface KSPropertyDeclaration : KSDeclaration {
     fun isDelegated(): Boolean
 
     /**
-     * Find the original overridee of this property, if overriding.
-     * @return [KSPropertyDeclaration] for the original property, if overriding, otherwise null.
+     * Find the closest overridee of this property, if overriding.
+     *
+     * For the following input:
+     * ```
+     * abstract class A {
+     *   open val x:Int
+     *   open val y:Int
+     * }
+     * abstract class B : A() {
+     *   override val x:Int
+     * }
+     * abstract class C : B() {
+     *   override val x:Int
+     *   override val y:Int
+     * }
+     * ```
+     * Calling `findOverridee` on `C.x` will return `B.x`.
+     * Calling `findOverridee` on `C.y` will return `A.y`.
+     *
+     * When there are multiple super classes / interfaces with the property, the closest declaration
+     * to the current containing declaration is selected. If they are in the same level, the
+     * property of the first specified interface (in source) will be returned.
```

Comment:
nit: `specified super class or interface` instead of `specified interface`

Reply:
> The overriding for properties usually get complicated when it involves Java, can you add a test case where Java code is overriding a property declared in Kotlin code? See [this](https://github.com/google/ksp/blob/638d9a19a9880056c8c95a5a2f5775045cb026ae/compiler-plugin/testData/api/checkOverride.kt#L110) for example.

the findOverridee in the java implementation of property is not implemented (just returns null). I'm guessing that is because java fields do not override parents.

You mean there is a base kotlin class with `val x:T` and java overrides it with `getX():T`?
I'll give it a try but that makes me think, that would mean the `findOverridee` of `KSFunction` would need to return a `KSPropertyAccessor` that does not implement `KSFunction` :/. 
I'll see what happens there.

There was also some work discussed before where KSP right now does not properly resolve property accessors in java source code. If we do, then we should also implement findOverridee in KSPropertyDeclaration.

Reply:
actually trying to reproduce that hits this issue:
https://github.com/google/ksp/issues/94

Reply:
i'm actually able to workaround that issue by making KSJavaFileImpl use descriptors for the class instead of Java impls.
Java impls are causing lots of inconsistencies hence created this issue:
https://github.com/google/ksp/issues/177

Reply:
Thanks for looking into this so fast. 

Question: is the merge of this PR blocked by the progress on #177? Or can this be merged already? It would help me a lot: until this is fixed, I need to maintain my fork. I use Kotlin only. :-)

Reply:
@neetopia wdyt? I think this one is fine to merge as it should be an improvement over what we have (fixes kotlin properties, java is still broken).
#177 probably won't happen or if it happens it will take a long time.

Reply:
Sounds good, I'll merge this one.

Reply:
Thanks for the merge. Is it possible to create a release that contains this? I am using KSP as a dependency.
If not, what is a likely date for a new release, given your release schedule?

Reply:
I'll do a release this Friday PST.